components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCPermissionBasedInternalScopeValidator.java

```diff
@@ -242,7 +242,7 @@ Note that if there is association between a federated user and local user () 'As
                     PERMISSION_ROOT)) {
                 return new ArrayList<>(allScopes);
             } else if (ArrayUtils.contains(allowedResourcesForUser, ADMIN_PERMISSION_ROOT)) {
-                return new ArrayList<>(getAdminAllowedScopes(allScopes, requestedScopes));
+                return new ArrayList<>(getAdminAllowedScopes(allScopes, requestedScopes, isSystemScope));
             }
```

Comment:
what if we remove this whole else if block?

Reply:
It solves this problem and unit tests also got passed after removing that block. But isn't it risky to simply remove a code section?

Reply:
Not if we understand the consequences :)

We have used the if - else if block to short circuit individual permission checks when SYSTEM scope is presented.
Can we take a overall look at this method and make sure we don't have similar glitches.

For example why are we returning all the all scopes even for a super tenant user with root permissions without considering the requested scopes? (If block)
don't we need to consider the requested scopes even for a user who has root permission